omod/src/main/java/org/openmrs/module/sync2/page/controller/Sync2PageController.java

```diff
@@ -0,0 +1,30 @@
+package org.openmrs.module.sync2.page.controller;
+
+import org.apache.commons.lang.StringUtils;
+import org.openmrs.module.sync2.api.utils.SyncUtils;
+import org.openmrs.module.uicommons.util.InfoErrorMessageUtil;
+import org.openmrs.ui.framework.UiUtils;
+import org.openmrs.ui.framework.page.PageModel;
+import org.springframework.stereotype.Controller;
+
+import javax.servlet.http.HttpSession;
+
+@Controller
+public class Sync2PageController {
+
+	private static final String VALID_PARENT_INSTANCE_URI = "validParentInstanceUri";
+
+	private static final String PARENT_URL_ERROR = "sync2.configuration.parentUrl.notValid";
+
+	public void controller(PageModel model, HttpSession session, UiUtils ui) {
+		boolean isValidParentInstanceUrl = parentInstanceUriIsValid();
+		model.addAttribute(VALID_PARENT_INSTANCE_URI, isValidParentInstanceUrl);
+		if (!isValidParentInstanceUrl) {
+			InfoErrorMessageUtil.flashErrorMessage(session, ui.message(PARENT_URL_ERROR));
+		}
+	}
+
+	private boolean parentInstanceUriIsValid() {
+		return StringUtils.isNotBlank(SyncUtils.getParentBaseUrl());
```

Comment:
Maybe it is possible to validate the url more strictly using some java classes?

Reply:
In the scope of this ticket I should verify if URL is not empty. However when we have set not valid parent URL address and then try to manual pull from parent we get the message about failure. But when we try to manual push to parent then we get the message about success. 

So we can create the new ticket for moving the same logic (which verify the URL before executing operation) from pull operation to push operation. @oskarhinc @kmadej what do you think about that?

The message after manual pull:
![image](https://user-images.githubusercontent.com/29865902/46427540-fe6a9680-c741-11e8-807c-289acfeaff3e.png)

The message after manual push:
![image](https://user-images.githubusercontent.com/29865902/46427650-3b368d80-c742-11e8-9613-10967c639f9b.png)

Reply:
@areklalo yes, please create a separate ticket for it

Reply:
I created the ticket: https://issues.openmrs.org/browse/SYNCT-179